zproject/backends.py

```diff
@@ -1655,6 +1661,33 @@ def decode_id_token(self, id_token: str) -> Dict[str, Any]:
 
         return decoded
 
+    def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
+        if not self.is_native_flow():
+            # The default implementation in python-social-auth is the browser flow.
+            return super().auth_complete(*args, **kwargs)
+
+        # We handle the native flow on our own.
```

Comment:
Can you expand this to give more discussion of the high-level strategy.  I see we go towards calling `do_auth`, like the original python-social-auth code does ... what is the intent of our changes from the base logic here?

Reply:
Posted a larger block comment detailing things